patches/minecraft/net/minecraft/server/integrated/IntegratedServer.java.patch

```diff
@@ -57,7 +57,56 @@
      }
  
      public void func_71217_p()
-@@ -213,6 +228,7 @@
+     {
++        if(this.field_71349_l.integratedServerCrash != null) return;
+         boolean flag = this.field_71348_o;
+         this.field_71348_o = Minecraft.func_71410_x().func_147114_u() != null && Minecraft.func_71410_x().func_147113_T();
+ 
+@@ -167,37 +187,56 @@
+         }
+         else
+         {
++            try
++            {
+             super.func_71217_p();
+ 
+             if (this.field_71349_l.field_71474_y.field_151451_c != this.func_184103_al().func_72395_o())
+@@ -198,6 +216,34 @@
+                     }
+                 }
+             }
++            }
++            catch(Throwable t) //Forge: just close the world instead of stopping the whole game
++            {
++                CrashReport worldcrash;
++                if(t instanceof net.minecraft.util.ReportedException) {
++                    worldcrash = ((net.minecraft.util.ReportedException)t).func_71575_a();
++                }else {
++                    worldcrash = new CrashReport("Unexpected error", t);
++                }
++                this.func_71230_b(worldcrash);
++                field_147148_h.fatal("Exception thrown in server world tick!", t);
++                net.minecraft.init.Bootstrap.func_179870_a(worldcrash.func_71502_e());
++                
++                for (EntityPlayerMP entityplayermp : Lists.newArrayList(IntegratedServer.this.func_184103_al().func_181057_v()))
++                {
++                    IntegratedServer.this.func_184103_al().func_72367_e(entityplayermp);
++                }
++                
++                super.func_71263_m();
++                
++                try {
++                    Thread.sleep(100L);
```

Comment:
Why are you doing this? Or well, anything in this handler. Could use some explaining.

Reply:
I'd have to change a bunch of netclienthandler stuff to supress the server closed message so instead I just delayed setting the variable so the code that displays the error message gets triggered well after it gets disconnected

Reply:
Ya, that isn't gunna fly, it just screams race condition. There needs to be a more robust method to do this.

Reply:
I'll mess around a bit but I can't guarantee the patches will stay this small.